pkg/types/dsse/dsse.go

```diff
@@ -15,23 +15,49 @@
 package dsse
 
 import (
+	"crypto/sha256"
 	"fmt"
 
+	v1 "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 	pb "github.com/sigstore/rekor-tiles/pkg/generated/protobuf"
 	"github.com/sigstore/rekor-tiles/pkg/types/verifier"
 )
 
-func Validate(ds *pb.DSSERequest) error {
-	if ds.Envelope == "" {
-		return fmt.Errorf("missing envelope")
+func ToLogEntryV0_0_2(ds *pb.DSSERequestV0_0_2) (*pb.DSSELogEntryV0_0_2, error) {
+	if ds.Envelope == nil {
+		return nil, fmt.Errorf("missing envelope")
 	}
-	if len(ds.Verifier) == 0 {
-		return fmt.Errorf("missing verifiers")
+	if len(ds.Verifiers) == 0 {
+		return nil, fmt.Errorf("missing verifiers")
 	}
-	for _, v := range ds.Verifier {
+	for _, v := range ds.Verifiers {
 		if err := verifier.Validate(v); err != nil {
-			return err
+			return nil, err
 		}
 	}
-	return nil
+
+	if ds.Envelope.Payload == nil {
+		return nil, fmt.Errorf("missing envelope payload")
+	}
+	payloadHash := sha256.Sum256([]byte(ds.Envelope.Payload))
+
+	// TODO: check that signatures can be validated against the provided keys
```

Comment:
I'm working on this in https://github.com/sigstore/rekor-tiles/issues/10 in case you haven't seen it

Reply:
ah okay cool. I'll mark the TODO with `#10` on it.